crucible-llvm/src/Lang/Crucible/LLVM/Intrinsics/LLVM.hs

```diff
@@ -1288,6 +1341,113 @@ llvmX86_SSE2_storeu_dq =
   [llvmOvr| void @llvm.x86.sse2.storeu.dq( i8*, <16 x i8> ) |]
   (\memOps args -> Ctx.uncurryAssignment (callStoreudq memOps) args)
 
+
+-- | Build an 'LLVMOverride' for a vector reduce intrinsic.
+llvmVectorReduce ::
+     (1 <= intSz)
+  => String
+     -- ^ The name of the operation to reduce (@add@, @mul@, etc.).
+  -> (forall r args ret
+       . IsSymInterface sym
+      => NatRepr intSz
+      -> RegEntry sym (VectorType (BVType intSz))
+      -> OverrideSim p sym ext r args ret (SymBV sym intSz))
+     -- ^ The semantics of the override.
+  -> NatRepr vecSz
+     -- ^ The size of the vector type.
+  -> NatRepr intSz
+     -- ^ The size of the integer type.
+  -> LLVMOverride p sym ext
+        (EmptyCtx ::> VectorType (BVType intSz))
```

Comment:
It may be useful in the future to create a `VectorSizedType vecSz (BVType intSz)` representation to correlate/constrain the arguments, but I don't think it's worth the impact to this PR.